lib/generator.js

```diff
@@ -136,6 +142,25 @@ class Generator {
     });
   }
 
+  /**
+    * Check if the Registry Options are valid or not.
+    * 
+    * @private
+    * @param {Object} invalidRegOptions Invalid Registry Options.
+    * 
+    */
+
+  verifyOptions(Options) {
+    if (typeof Options !== 'object') return [];  
+    let invalidOptions = Object.keys(Options).filter(param => !GENERATOR_OPTIONS.includes(param)); 
+    if (Options.registry) {
+      invalidOptions = Object.keys(Options.registry).filter(param => !GENERATOR_OPTIONS.includes(param));
```

Comment:
but this step will override what already is listed in `invalidOptions` in line 155? just modify `throws an error indicating an unexpected param was given for registry configuration` test by passing some invalid option, not `registry` related and I bet error will still point `privateKey` only

Reply:
@derberg Changed